src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -104,15 +104,16 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None):
         self.dim = dim
         self.max_position_embeddings = max_position_embeddings
         self.base = base
-        self.register_buffer("inv_freq", None, persistent=False)
```

Comment:
I hope there is no specific (important enough) reason to do so.

Reply:
Do you mean we want to keep it always in fp32, but with buffer + if we give it values at __init__, it could be changed to other dtype at later stage of `from_pretrained`?

Reply:
If this is the case, I would suggest let's first reach to an agreement that the issue of current implementation of  `self.inv_freq` doesn't work (actually) with `torch.compile`

It would be nice if you can try the provided code snippet. And if you want some proof from the test `test_torch_compile_fullgraph`, I can work on that too.

Then we discuss what would be a better fix.